sql/core/src/test/resources/sql-tests/inputs/postgreSQL/numeric.sql

```diff
@@ -884,6 +884,12 @@ SELECT '' AS to_char_10, to_char(val, 'S0999999999999999.999999999999999'), val
 -- SELECT '' AS to_char_35, to_char('100'::numeric, 'f"ool\"999');
 -- SELECT '' AS to_char_36, to_char('100'::numeric, 'f"ool\\"999');
 
+-- [SPARK-XXX] Add support for additional SQL functions
+-- TO_VARCHAR()
+-- Alias of TO_CHAR()
+SELECT '' AS to_varchar_3, to_varchar(val, '9999999999999999.999999999999999PR'), val
```

Comment:
I think you will also need to get analyzer golden file results as well. 

![image](https://github.com/apache/spark/assets/99207096/3696db22-c1d2-4a33-a67e-2c8ed346b03e)